Add decky-syncthing

theCapypara opened the pull request:
<!-- Make sure to include your plugin name below! -->

# decky-syncthing (Syncthing)

> Plugin for managing Syncthing to synchronize files with other devices. Not officially affiliated with the Syncthing project.

From the README:

> A [Decky](https://decky.xyz/) plugin to manage [Syncthing](https://syncthing.net/) from your Steam Deck.
>
> This plugin allows you to start and stop Syncthing from the Quick Access panel of your Steam Deck.
>
> It also has some basic stats and allows you to access the web UI. It can also be configured to automatically start Syncthing with Gamescope (the Steam Deck main UI).
>
> The plugin works no matter if you have HTTPS enabled or not and also no matter if you have basic auth enabled.
>
> Since the Steam Deck UI has no support for self-signed HTTPS certificates or Basic Auth, this plugin starts a proxy server on port 58384 (HTTP, localhost only) that forwards requests to the Syncthing Web UI and API. 

![Screenshot of the plugin](https://raw.githubusercontent.com/theCapypara/steamdeck-decky-syncthing/main/banner.png)

**NOTE: This PR template contained a link asking me to review https://deckbrew.xyz/en/plugin-dev/review-and-testing - this page does not exist! So I did not review that page.**

## Checklist:

### Developer Checklist

- [X] I am the original author or an authorized maintainer of this plugin.
- [X] I have abided by the licenses of the libraries I am utilizing, including attaching license notices where appropriate.

### Plugin Checklist

- [X] I have verified that my plugin works properly on the Stable and Beta update channels of SteamOS.
- [X] I have verified my plugin is unique or alternatively provides more/alternative functionality to a similar plugin already on the store.

### Plugin Backend Checklist

- **Yes**: I am using a custom backend other than Python.
- **Yes**: I am using a tool or software from a 3rd party FOSS project that does not have it's dependencies [statically linked](https://en.wikipedia.org/wiki/Static_library).
- **No**: I am using a custom binary that has all of it's dependencies statically linked.

The Rust backend is mostly statically linked but does rely on some very standard system libs like OpenSSL 1.1.

## Testing

- [ ] Tested on SteamOS Preview Update Channel.

theCapypara commented:
Found the review and testing page at https://wiki.deckbrew.xyz/en/plugin-dev/review-and-testing - will review that now!

theCapypara commented:
Changed banner to be a PNG.

PartyWumpus commented:
You should probably have something in the plugin that mentions that the syncthing flatpak has to be installed, as users are likely to go in with zero knowledge. 

The banner image needs to be a png iirc, so if you could change then it would likely work.

Also I can see you're using the custom scrollable element, hopefully that can be replaced by scrollGroup and friends when https://github.com/SteamDeckHomebrew/decky-frontend-lib/pull/90 gets merged.

theCapypara commented:
We noticed that OpenSSL 1.1 was apparently not guaranteed to be installed on a Deck, so I'm switching to a statically linked TLS implementation.

theCapypara commented:
I replaced the link to OpenSSL with the statically compiled `rustls` so it should work now. I tested it on my deck and confirmed the executable does not link against OpenSSL anymore.

Additionally I updated the README and the description in the `plugin.json` to clarify that the Flatpak needs to already be installed:

PartyWumpus commented:
Sorry I took so long to get back around to this. The plugin mostly functions well (entering the API key was a little tedious). However, the actual disabling/enabling of folders doesn't seem to work, they don't do anything when clicked on.

(ignore the errors in the console, they seem unrelated)

https://github.com/SteamDeckHomebrew/decky-plugin-database/assets/48649272/ae437b34-144b-4034-bd2b-c1843933bd00

theCapypara commented:
For now that's on purpose. Eventually I'll probably make it so that clicking the folders and devices shows some details.
For now representing them as buttons seemed visually to be the nicest way of doing it, even if the buttons don't do anything.

PartyWumpus commented:
Ah okay, guess it's all working fine then.

PartyWumpus reviewed (approved):
lgtm
